pkg/internalmetrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+package internalmetrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// These constants define the different label names for the different metric timeseries
+const (
+	labelVersion      = "version"
+	labelNamespace    = "namespace"
+	labelSMCPVersion  = "smcp_version"
+	labelSMCPTopology = "smcp_topology"
+)
+
+// MetricsType defines all of istio-operator's metrics.
+type MetricsType struct {
+	SMCPCounter *prometheus.CounterVec
```

Comment:
This should be called something like "MemberCount(er?)", not SMCPCount(er), since it doesn't count the number of SMCPs, but the number of member namespaces.